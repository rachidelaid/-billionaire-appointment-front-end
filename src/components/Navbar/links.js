const links = [
  {
    path: '/',
    description: 'BILLIONAIRES',
    permission: 1,
  },
  {
    path: '/appointment/new',
    description: 'RESERVE BILLIONAIRES',
    permission: 2,
  },
  {
    path: '/appointments',
    description: 'MY APPOINTMENTS',
    permission: 2,
  },
  {
    path: '/billionaire/new',
    description: 'ADD BILLIONAIRE',
    permission: 3,
  },
  {
    path: '/billionaire/delete',
    description: 'DELETE BILLIONAIRE',
    permission: 3,
  },
];

export default links;
