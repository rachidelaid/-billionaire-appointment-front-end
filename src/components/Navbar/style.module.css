.flex-center {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.nav {
  height: 100%;
  width: 200px;
  position: fixed;
  z-index: 1;
  top: 0;
  left: 0;
  overflow-x: hidden;
  transition: 0.5s;
  border-right: 3px solid #f8f8f8;
  margin-left: 1rem;
}

.logo {
  height: 200px;
}

.list {
  list-style: none;
}

.links {
  text-align: center;
  color: var(--black);
}

.links li,
.link {
  width: 100%;
}

.link {
  padding: 0.7rem;
  font-family: var(--font-bold);
  font-size: 1.3rem;
  display: block;
}

.link:hover {
  background-color: var(--orange);
  color: #fff;
}

.nav-footer {
  display: flex;
  flex-direction: column;
  color: var(--black);
  margin-top: auto;
  padding-right: 1rem;
  gap: 1rem;
}

.social {
  display: flex;
  justify-content: center;
  gap: 1rem;
  font-size: 1.2rem;
}

.copyright {
  font-size: 0.7rem;
  margin-bottom: 1rem;
}
