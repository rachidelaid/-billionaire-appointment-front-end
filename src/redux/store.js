import { configureStore } from '@reduxjs/toolkit';
import userReducer from './users';
import billionaireReducer from './billionaires';

const store = configureStore({
  reducer: {
    billionaires: billionaireReducer,
    users: userReducer,
  },
});

export default store;
