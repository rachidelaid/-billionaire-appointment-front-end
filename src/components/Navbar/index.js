import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import style from './style.module.css';
import links from './links';

const Navbar = () => {
  const [isNavExpanded, setIsNavExpanded] = useState(false);
  const currentUserRole = useSelector((state) => (state.users.user ? state.users.user.role : 'everyone'));

  const activeStyle = {
    backgroundColor: 'var(--green)',
    color: 'white',
  };

  return (
    <div className={style.container}>
      <button
        type="button"
        onClick={() => {
          setIsNavExpanded(!isNavExpanded);
        }}
        className={
          isNavExpanded ? style.hidden : style.hamburger
        }
      >
        <i className="bi bi-list" />
      </button>
      <nav className={`${style.nav} ${style['flex-center']} ${!isNavExpanded && style.hidden}`}>
        <button
          type="button"
          onClick={() => {
            setIsNavExpanded(!isNavExpanded);
          }}
          className={style['nav-hide']}
        >
          <i className="bi bi-caret-left" />
        </button>
        <div className={style['logo-container']}>
          <img
            className={style.logo}
            src="https://raw.githubusercontent.com/orozCoding/billionares-pictures/main/logo/billionaires_logo.png"
            alt="Billionaires Appointments logo"
          />
        </div>
        <ul className={`${style.links} ${style['flex-center']} ${style.list}`}>
          {links.map(({ path, description, permission }) => (
            permission.includes(currentUserRole) && (
            <li key={description}>
              <NavLink
                className={style.link}
                to={path}
                style={({ isActive }) => (isActive ? activeStyle : undefined)}
              >
                {description}
              </NavLink>
            </li>
            )
          ))}
        </ul>
        <div className={style['nav-footer']}>
          <ul className={`${style.links} ${style['flex-center']} ${style.list}`}>
            <li>
              <NavLink
                className={`${style.link} ${style.logout}`}
                to="/logout"
              >
                LOGOUT
              </NavLink>
            </li>
          </ul>
          <ul className={`${style.social} ${style.list}`}>
            <li>
              <a href="https://twitter.com">
                <i className="bi bi-twitter" />
              </a>
            </li>
            <li>
              <a href="https://about.facebook.com/meta/">
                <i className="bi bi-meta" />
              </a>
            </li>
            <li>
              <a href="https://myaccount.google.com/">
                <i className="bi bi-google" />
              </a>
            </li>
            <li>
              <a href="https://vimeo.com/">
                <i className="bi bi-vimeo" />
              </a>
            </li>
            <li>
              <a href="https://www.pinterest.com/">
                <i className="bi bi-pinterest" />
              </a>
            </li>
          </ul>
          <small className={style.copyright}>Â© 2022 BOAS & R.A.M.P - RAMP</small>
        </div>
      </nav>
    </div>
  );
};
export default Navbar;
