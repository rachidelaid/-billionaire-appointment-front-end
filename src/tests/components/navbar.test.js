import {
  fireEvent,
  render, screen, within,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Navbar from '../../components/Navbar';
import store from '../../redux/store';

const Wrapper = () => (
  <Provider store={store}>
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>
  </Provider>
);

describe('Navbar Component', () => {
  it('should render logo image', () => {
    render(<Wrapper />);
    const logoImg = screen.getByRole('img', {
      name: /billionaires appointments logo/i,
    });

    expect(logoImg).toMatchSnapshot();
  });

  it('should render only billionaires link if user is not logged in', () => {
    render(<Wrapper />);

    const billionairesLink = screen.getByRole('link', {
      name: /billionaires/i,
    });

    const signinLink = screen.getByRole('link', {
      name: /sign in/i,
    });

    expect(billionairesLink).toMatchSnapshot();
    expect(signinLink).toMatchSnapshot();
  });

  it('should show nav', () => {
    render(<Wrapper />);

    const navigation = screen.getByRole('navigation');
    const hideNavBtn = within(navigation).getByRole('button');

    expect(hideNavBtn).toMatchSnapshot();

    fireEvent.click(hideNavBtn);

    expect(navigation.style.transform).toEqual('translateX(-100%)');
  });
});
