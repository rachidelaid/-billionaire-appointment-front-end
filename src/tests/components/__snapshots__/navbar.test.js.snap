// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should hide nav button 1`] = `
<button
  class="nav-hide"
  type="button"
>
  <i
    class="bi bi-caret-left"
  />
</button>
`;

exports[`Navbar Component should render logo image 1`] = `
<img
  alt="Billionaires Appointments logo"
  class="logo"
  src="https://raw.githubusercontent.com/orozCoding/billionares-pictures/main/logo/billionaires_logo.png"
/>
`;

exports[`Navbar Component should render only billionaires link if user is not logged in 1`] = `
<a
  aria-current="page"
  class="link active"
  href="/"
  style="color: white;"
>
  BILLIONAIRES
</a>
`;

exports[`Navbar Component should render only billionaires link if user is not logged in 2`] = `
<a
  class="link"
  href="/login"
>
  Sign In
</a>
`;

exports[`Navbar Component should show nav 1`] = `
<button
  class="nav-hide"
  type="button"
>
  <i
    class="bi bi-caret-left"
  />
</button>
`;
